### PR TITLE
Clean up Amiga build process

### PR DESCRIPTION
**Pull request by AJenbo**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,14 +6,6 @@
 # macOS
 .DS_Store
 
-# qt-creator
```
Comment (AJenbo) [inline .gitignore:9]: People should set up global ignores, I'm going to clean up this file soon it's impossible to look at as is.

File: .gitignore [hunk 2]
```diff
@@ -428,18 +420,8 @@ DerivedData/
 **/xcshareddata/WorkspaceSettings.xcsettings
 
 /libs
+
 # End of https://www.gitignore.io/api/xcode
 
 # Don't accidently commit the diabdat.mpq
 *.mpq
-
-# Static libs, generated files and directories
```
Comment (AJenbo) [inline .gitignore:436]: Putting a .gitignore with `*` in the build folder should do, might even commit that.

File: CMakeLists.txt [hunk 3]
```diff
@@ -279,28 +249,21 @@ if(NOT NONET)
...
-    zlib)
-  if(WARPOS)
-    target_link_libraries(devilutionx PRIVATE -lmikmod)
+if(AMIGA)
+  if(USE_SDL1)
+    target_link_libraries(devilutionx PRIVATE
+      ${SDL_LIBRARY}
+      ${SDL_TTF_LIBRARY}
+      ${SDL_MIXER_LIBRARY}
+      ${FREETYPE_LIBRARIES})
```
Comment (AJenbo) [inline CMakeLists.txt:258]: For some obscure reason, I can't just add this to the for each, or by itself?!?
(SDL_LIBRARY, SDL_TTF_LIBRARY, SDL_MIXER_LIBRARY are already set to be linked just a few lines further down)

File: Jenkinsfile [hunk 4]
```diff
@@ -1,369 +0,0 @@
-def notify(status){
```
Comment (AJenbo) [inline Jenkinsfile:1]: Be sure to make a back up :)

**Comment by AJenbo** [comment]
Test build for this PR: [devilutionx.zip](https://github.com/AmigaPorts/devilutionX/files/3671988/devilutionx.zip)

File: Jenkinsfile (hunk 4 above)
Comment (MBeijer) [reply Jenkinsfile:1]: It's still in the master-backup branch (and commit history)

File: .circleci/config.yml [hunk 5]
```diff
@@ -71,8 +71,8 @@ jobs:
       - checkout
       - run: Packaging/amiga/prep.sh
       - run: mkdir build
-      - run: cd build && PKG_CONFIG_PATH=/opt/m68k-amigaos/lib/pkgconfig/:/opt/m68k-amigaos/share/pkgconfig/ cmake -DM68K_CPU=68040 -DM68K_FPU=hard -DM68K_COMMON="-s -ffast-math -O3 -noixemul -fpermissive -DAMIGA -fbbb=abcdef" ..
-      - run: cd build && cmake --build . --config Release -- -j$(nproc)
+      - run: cd build && PKG_CONFIG_PATH=/opt/m68k-amigaos/lib/pkgconfig/:/opt/m68k-amigaos/share/pkgconfig/ cmake -DM68K_CPU=68040 -DM68K_FPU=hard -DM68K_COMMON="-s -ffast-math -O3 -noixemul -DBIG_ENDIAN -DAMIGA -fpermissive -fbbb=abcdef" ..
+      - run: cd build && make -j$(nproc)
```
Comment (MBeijer) [inline .circleci/config.yml:75]: Why make?

File: .gitignore (hunk 1 above)
Comment (MBeijer) [reply .gitignore:9]: some people managed to add their whole build dir and commit that. this was a precaution

File: CMakeLists.txt (hunk 3 above)
Comment (MBeijer) [reply CMakeLists.txt:258]: when linking static libraries, the link order is crucial. As an example, freetype needs to be linked in _after_ SDL_TTF

**Review by MBeijer — APPROVED** [review]

File: .circleci/config.yml (hunk 5 above)
Comment (AJenbo) [reply .circleci/config.yml:75]: It's what we do on all other builds, it's the default generator and it's simple to read and understand for most users that want to try and compile them selfs.

Comment (MBeijer) [reply .circleci/config.yml:75]: Well, so is cmake --build, and on systems with msbuild, it will default to that instead of GNU Make.

Comment (AJenbo) [reply .circleci/config.yml:75]: Noted I might update all the guides soon and will probably apply this change in the process, thanks

Comment (AJenbo) [reply .circleci/config.yml:75]: Everything has now been switched to recomend cmake --build instead https://github.com/diasurgical/devilutionX/commit/ebe846d6ef279b7977dc87e0ef0e43ac4661e891

**Pull request merged**